pkg/server/singleprocess/service_ui_deploy.go

```diff
@@ -5,15 +5,54 @@ import (
 	"fmt"
 	"strings"
 
+	"github.com/hashicorp/go-hclog"
+	"github.com/pkg/errors"
+
 	pb "github.com/hashicorp/waypoint/pkg/server/gen"
 	"github.com/hashicorp/waypoint/pkg/server/ptypes"
 	"github.com/hashicorp/waypoint/pkg/serverstate"
 )
 
+func (s *Service) UI_GetDeployment(
+	ctx context.Context,
+	req *pb.UI_GetDeploymentRequest,
+) (*pb.UI_GetDeploymentResponse, error) {
+	log := hclog.FromContext(ctx)
+
+	getDeployReq := &pb.GetDeploymentRequest{Ref: req.Ref, LoadDetails: req.LoadDetails}
+
+	deploy, err := s.GetDeployment(ctx, getDeployReq)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting deployment %q", deploy.Id)
+	}
+
+	bundle, err := s.getDeploymentBundle(ctx, deploy.Application, deploy.Workspace, deploy)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error getting bundle for deployment %q", deploy.Id)
+	}
+
+	if bundle.Deployment.HasEntrypointConfig {
+		bundle.DeployUrl, err = s.getDeployUrl(ctx, deploy)
+		if err != nil {
+			log.Warn(
+				"failed getting horizon hostname for deployment, but will attempt again for next deployment",
```

Comment:
Referencing the "next deployment" doesn't make sense in the context of this "UI_GetDeployment" function.

Reply:
Makes sense! I removed the `but will attempt again for next deployment` bit.